### PR TITLE
Feature/cc backend items from pk

### PR DESCRIPTION
**Pull request by OpaJaa**
Working parts:
- get items list from pk
- get item info from pk

Changed item info -path, so that it is easier to know what to do with the request.

Code from PK-service pull request is used here so this branch should work. You need to add .env -files with your credentials to both services. Examples should be in there. Need to resolve conflicts when merging or pull all the needs stuff when pk-service pr is merged.

Need help in organizing these in a good fashion. Also need help with refactoring code. Authentication is also something I don't really know what to do with, but I added the token and id fetching from Hanna's code to backend also so the token could be used with the requests. I assume this should come from the ui later on?

All comments are welcome and don't hesitate to criticize! I know this is not a good solution so I can write everything from the start if necessary. Just don't have the time now... :D

### CODE REVIEW

**Review by bunjovis — COMMENTED** [review]
At least utils.ts, types.ts, itemRoutes.ts, systemRoutes.ts, userRoutes.ts have some indentation problems. Some parts are missing indentation and there are also mixed indentations: you should only use one, 2 or 4 spaces indentation. Some parts of code have extra indentation in front of them, when they should not have any indentation. I think we need to setup eslint correctly for each service to get these styling issues in check. Otherwise, seems good at the moment, at least to me. You're correct that the PK token would come from the UI in the future.

**Comment by OpaJaa** [comment]
> At least utils.ts, types.ts, itemRoutes.ts, systemRoutes.ts, userRoutes.ts have some indentation problems. Some parts are missing indentation and there are also mixed indentations: you should only use one, 2 or 4 spaces indentation. Some parts of code have extra indentation in front of them, when they should not have any indentation. I think we need to setup eslint correctly for each service to get these styling issues in check. Otherwise, seems good at the moment, at least to me. You're correct that the PK token would come from the UI in the future.

EDIT. Found Prettier -extension and now using that. Should make things easier.

Ah yes, this has been a problem. Especially when moving code to other places with cut/paste, vs code seems to do some weird stuff with indentations. I'll try to correct them. Does anyone have any idea if vs code can be used with some indentation rules that would be applied with every save etc? 

I haven't setup eslint at all. So some setup instructions for that would be nice. For now I don't have the time to do that as I need to write my bachelors thesis 😁 😬

File: cc-backend/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import * as OpenApiValidator from "express-openapi-validator";
+import sysRouter from "./routes/systemRoutes";
+import itemRouter from "./routes/itemRoutes";
+import advertRouter from "./routes/advertRoutes";
+import userRouter from "./routes/userRoutes";
+
+import { Error } from "./types";
+import { Item } from "./types";
+import { UserInfo } from "./types";
+import { Config } from "./types";
```
Comment (olivena) [inline cc-backend/src/index.ts:15]: I think these can be written as:
import {
  Error,
  Item,
  UserInfo,
  Config} from "./types";

**Review by olivena — DISMISSED** [review]

**Review by bunjovis — DISMISSED** [review]

File: cc-backend/src/index.ts (hunk 1 above)
Comment (OpaJaa) [reply cc-backend/src/index.ts:15]: That is true and I've used that in other places. I will fix these later as I hopefully have more time next week. Some imports are also not needed so gotta check them too.

**Comment by OpaJaa** [comment]
Pulled changes from main and hopefully resolved the conflicts correctly.

**Review by bunjovis — APPROVED** [review]

**Review by hannakortetjarvi — APPROVED** [review]

**Pull request merged**